osu.Game/Overlays/BeatmapListingOverlay.cs

```diff
@@ -112,27 +123,51 @@ private void onSearchStarted()
 
         private void onSearchFinished(List<BeatmapSetInfo> beatmaps)
         {
+            //No matches case
             if (!beatmaps.Any())
             {
-                LoadComponentAsync(new NotFoundDrawable(), addContentToPlaceholder, (cancellationToken = new CancellationTokenSource()).Token);
+                LoadComponentAsync(notFoundContent, addContentToPlaceholder, (cancellationToken = new CancellationTokenSource()).Token);
                 return;
             }
 
-            var newPanels = new FillFlowContainer<BeatmapPanel>
+            //New query case
+            if (!shouldAddNextPage)
```

Comment:
flimsy as hell. try scrolling while you are fetching more. it'll replace existing panels.